src/ImageSharp/Processing/Processors/Binarization/AdaptiveThresholdProcessor{TPixel}.cs

```diff
@@ -0,0 +1,165 @@
+                        ref TPixel rowRef = ref MemoryMarshal.GetReference(row);
+                        ref TPixel color = ref Unsafe.Add(ref rowRef, x);
+                        color.ToRgba32(ref rgb);
+
+                        sum += (ulong)(rgb.R + rgb.G + rgb.G);
+                        if (x - startX != 0)
+                        {
+                            intImage[x - startX, y - startY] = intImage[x - startX - 1, y - startY] + sum;
+                        }
+                        else
+                        {
+                            intImage[x - startX, y - startY] = sum;
+                        }
+                    }
+                }
+
+                var operation = new RowOperation(workingRectangle, source, intImage, upper, lower, thresholdLimit, clusterSize, startX, endX, startY);
+                ParallelRowIterator.IterateRows(
+                    configuration,
+                    workingRectangle,
+                    in operation);
+            }
+        }
+
+        private readonly struct RowOperation : IRowOperation
+        {
+            private readonly Rectangle bounds;
+            private readonly ImageFrame<TPixel> source;
+            private readonly Buffer2D<ulong> intImage;
+            private readonly TPixel upper;
+            private readonly TPixel lower;
+            private readonly float thresholdLimit;
+            private readonly int startX;
+            private readonly int endX;
+            private readonly int startY;
+            private readonly byte clusterSize;
+
+            [MethodImpl(InliningOptions.ShortMethod)]
+            public RowOperation(
+                Rectangle bounds,
+                ImageFrame<TPixel> source,
+                Buffer2D<ulong> intImage,
+                TPixel upper,
+                TPixel lower,
+                float thresholdLimit,
+                byte clusterSize,
+                int startX,
+                int endX,
+                int startY)
+            {
+                this.bounds = bounds;
+                this.source = source;
+                this.intImage = intImage;
+                this.upper = upper;
+                this.lower = lower;
+                this.thresholdLimit = thresholdLimit;
+                this.startX = startX;
+                this.endX = endX;
+                this.startY = startY;
+                this.clusterSize = clusterSize;
+            }
+
+            /// <inheritdoc/>
+            [MethodImpl(InliningOptions.ShortMethod)]
+            public void Invoke(int y)
+            {
+                Rgba32 rgb = default;
+                Span<TPixel> pixelRow = this.source.GetPixelRowSpan(y);
+
+                for (int x = this.startX; x < this.endX; x++)
+                {
+                    TPixel pixel = pixelRow[x];
+                    pixel.ToRgba32(ref rgb);
+
+                    var x1 = (ushort)Math.Max(x - this.startX - this.clusterSize + 1, 0);
```

Comment:
This cast limits our range. If `x` or `y` are greater than `ushort.MaxValue` we end up with the wrong offset.

Reply:
somehow i managed to miss that